src/test/java/itest/ShouldAcceptMultipartWithBoundaryIT.java

```diff
@@ -0,0 +1,172 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package itest;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import io.cryostat.net.web.http.HttpMimeType;
+
+import io.netty.handler.timeout.TimeoutException;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.HttpException;
+import itest.bases.StandardSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class ShouldAcceptMultipartWithBoundaryIT extends StandardSelfTest {
+    static final String TEST_RULE_NAME = "Test_Rule";
+
+    static final Map<String, String> NULL_RESULT = new HashMap<>();
+
+    static {
+        NULL_RESULT.put("result", null);
+    }
+
+    // @BeforeAll
+    // static void setup() throws Exception {}
+
+    @AfterEach
+    void cleanup() throws Exception {
+        CompletableFuture<JsonObject> deleteResponse = new CompletableFuture<>();
+        webClient
+                .delete(String.format("/api/v2/rules/%s", TEST_RULE_NAME))
+                .send(
+                        ar -> {
+                            if (ar.succeeded()) {
+                                deleteResponse.complete(ar.result().bodyAsJsonObject());
+                            } else {
+                                deleteResponse.completeExceptionally(ar.cause());
+                            }
+                        });
+
+        JsonObject expectedDeleteResponse =
+                new JsonObject(
+                        Map.of(
+                                "meta",
+                                Map.of("type", HttpMimeType.JSON.mime(), "status", "OK"),
+                                "data",
+                                NULL_RESULT));
+
+        try {
+            JsonObject deleteResult = deleteResponse.get(5, TimeUnit.SECONDS);
+            MatcherAssert.assertThat(deleteResult, Matchers.equalTo(expectedDeleteResponse));
+        } catch (TimeoutException e) {
+            System.out.println("Deletion timed out. Reason: " + e.getMessage());
+            throw new RuntimeException("Deletion did not complete within the expected time.", e);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof HttpException) {
+                HttpException httpException = (HttpException) e.getCause();
+                MatcherAssert.assertThat(httpException.getStatusCode(), Matchers.equalTo(400));
+                MatcherAssert.assertThat(
```

Comment:
This is in the `@AfterAll` cleanup, not a test case, so I don't think there should be test assertions happening in here. If the case is expected to fail then that behaviour should be captured in the relevant `@Test`, not in the cleanup method.